### PR TITLE
[CARBONDATA-2357] Add column name and index mapping in Lucene datamap writer

### PR DESCRIPTION
**Pull request by jackylk**
In current LuceneDataMapWriter.onPageAdded, there is a loop based on string match which may not be efficient. This PR improves it by replacing it with a Map (constant access time)

 - [X] Any interfaces changed?
 No
 - [X] Any backward compatibility impacted?
 No
 - [X] Document update required?
No
 - [X] Testing done
run all testcase
 - [X] For large changes, please consider breaking it into sub-tasks under an umbrella JIRA. 
NA

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/3912/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/5133/

**Comment by jackylk** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/3917/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/5138/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/3932/

**Comment by jackylk** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/4024/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/5218/

File: datamap/lucene/src/main/java/org/apache/carbondata/datamap/lucene/LuceneDataMapWriter.java [hunk 1]
```diff
@@ -222,12 +236,10 @@ public void onPageAdded(int blockletId, int pageId, ColumnPage[] pages) throws I
...
 
-      // add other fields
-      for (int colIdx = 0; colIdx < columnsCount; colIdx++) {
-        if (indexedCarbonColumns.contains(pages[colIdx].getColumnSpec().getFieldName())) {
-          if (!pages[colIdx].getNullBits().get(rowId)) {
-            addField(doc, pages[colIdx], rowId, Field.Store.NO);
-          }
+      // add indexed columns value into the document
+      for (int colIdx = 0; colIdx < luceneColumnIndex.length; colIdx++) {
+        if (!pages[luceneColumnIndex[colIdx]].getNullBits().get(rowId)) {
```
Comment (QiangCai) [inline datamap/lucene/src/main/java/org/apache/carbondata/datamap/lucene/LuceneDataMapWriter.java]: The indexedCarbonColumns and pages have the same order. not required to check it again.

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/4089/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/5270/

**Comment by QiangCai** [comment]
LGTM

**Pull request closed without merging**